src/components/Chat/ChatItem.tsx

```diff
@@ -0,0 +1,130 @@
+import * as React from 'react'
+import * as PropTypes from 'prop-types'
+import * as cx from 'classnames'
+
+import {
+  createShorthandFactory,
+  customPropTypes,
+  IRenderResultConfig,
+  UIComponent,
+} from '../../lib'
+import {
+  ComponentPartStyle,
+  ComponentVariablesInput,
+  IComponentPartStylesPrepared,
+} from '../../../types/theme'
+import { Extendable, ItemShorthand, ReactChildren } from '../../../types/utils'
+import ChatBubble from './ChatBubble'
+import ChatAction from './ChatAction'
+import Divider from '../Divider/Divider'
+import childrenExist from '../../lib/childrenExist'
+
+export interface IChatItemProps {
+  action?: ItemShorthand
+  as?: any
+  bubble?: ItemShorthand
+  children?: ReactChildren
+  className?: string
+  divider?: ItemShorthand
+  styles?: ComponentPartStyle
+  variables?: ComponentVariablesInput
+}
+
+class ChatItem extends UIComponent<Extendable<IChatItemProps>, any> {
+  static className = 'ui-chat__item'
+
+  static create: Function
+
+  static displayName = 'ChatItem'
+
+  static propTypes = {
+    as: customPropTypes.as,
+
+    /** Shorthand for the bubble message. */
+    bubble: customPropTypes.itemShorthand,
+
+    /** Shorthand for the divider message. */
+    divider: customPropTypes.itemShorthand,
+
+    /** Shorthand for the control message. */
+    action: customPropTypes.itemShorthand,
+
+    /** Child content. */
+    children: PropTypes.node,
+
+    /** Additional classes. */
+    className: PropTypes.string,
+
+    /** Custom styles to be applied for component. */
+    styles: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
+
+    /** Custom variables to be applied for component. */
+    variables: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
+  }
+
+  static handledProps = [
+    'action',
+    'as',
+    'bubble',
+    'children',
+    'className',
+    'divider',
+    'styles',
+    'variables',
+  ]
+
+  static defaultProps = {
+    as: 'li',
+  }
+
+  renderComponent({
+    ElementType,
+    classes,
+    rest,
+    styles,
+    variables,
+  }: IRenderResultConfig<IChatItemProps>) {
+    const { children } = this.props
+
+    return (
+      <ElementType {...rest} className={cx(classes.root, classes.content)}>
+        {childrenExist(children) ? { children } : this.renderMessage(styles, variables)}
+      </ElementType>
+    )
+  }
+
+  private renderMessage = (
+    styles: IComponentPartStylesPrepared,
+    variables: ComponentVariablesInput,
+  ): React.ReactNode => {
+    const { bubble, action, divider } = this.props
+
+    return (
```

Comment:
right now we assume priorities `bubble` > `action` > `divider` and some users would be confused if they try to mix them (use `bubble` and `action` at the same time and only `bubble` renders); we need to make it obvious for user these are mutually exclusive, see my suggestion above

Reply:
Taken into consideration.